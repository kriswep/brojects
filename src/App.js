import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>&lt;BROJECTS&gt;</h2>
          <p>Better Github projects</p>
        </div>
        <p className="App-intro">
          Interactive and flexible project management on top of Github projects.
        </p>
      </div>
    );
  }
}

export default App;
