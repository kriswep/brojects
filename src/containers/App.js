import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  render() {
    return (
      <section className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>&lt;BROJECTS&gt;</h2>
          <p>Better Github projects</p>
        </header>
        <p className="App-intro">
          Interactive and flexible project management on top of Github projects.
        </p>
      </section>
    );
  }
}

export default App;
