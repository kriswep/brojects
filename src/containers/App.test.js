import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';

describe('App', () => {
  describe('rendering', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<App />, div);
    });

    it('renders a Header', () => {
      const wrapper = shallow(<App />);
      expect(wrapper.find('header').hasClass('App-header')).toBe(true);
    });
  });

  describe('mapStateToProps', () => {
    it('should pass auth object', () => {
      const fixture = {
        auth: {
          some: 'value',
        }
      }
      expect(mapStateToProps(fixture)).toMatchSnapshot();
    });
  });

  describe('mapDispatchToProps', () => {
    it('should handle onChange', () => {
      const dispatch = jest.fn();
      const dispatchedProps = mapDispatchToProps(dispatch);
      const fixture = {
        target: {
          value: 'something'
        }
      };
      dispatchedProps.onChange(fixture);
      expect(dispatch.mock.calls.length).toBe(1);
      expect(dispatch.mock.calls).toMatchSnapshot();
    });

    it('should handle onSubmit', () => {
      const dispatch = jest.fn();
      const preventDefault = jest.fn();
      const dispatchedProps = mapDispatchToProps(dispatch);
      const fixture = {
        preventDefault
      };
      dispatchedProps.onSubmit(fixture);
      expect(preventDefault.mock.calls.length).toBe(1);
      expect(dispatch.mock.calls.length).toBe(1);
      expect(dispatch.mock.calls).toMatchSnapshot();
    });

    it('should handle onClick', () => {
      const dispatch = jest.fn();
      const preventDefault = jest.fn();
      const dispatchedProps = mapDispatchToProps(dispatch);
      const fixture = {
        preventDefault
      };
      dispatchedProps.onClick(fixture);
      expect(preventDefault.mock.calls.length).toBe(1);
      expect(dispatch.mock.calls.length).toBe(3);
      expect(dispatch.mock.calls).toMatchSnapshot();
    });

  });

});
