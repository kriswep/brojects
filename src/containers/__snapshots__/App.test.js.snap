exports[`App mapDispatchToProps should handle onChange 1`] = `
Array [
  Array [
    Object {
      "authtoken": "something",
      "type": "brojects/action/SET_AUTHTOKEN",
    },
  ],
]
`;

exports[`App mapDispatchToProps should handle onClick 1`] = `
Array [
  Array [
    Object {
      "type": "brojects/action/GET_REPOS",
    },
  ],
  Array [
    Object {
      "repoFullName": "kriswep/modern-modular-javascript",
      "type": "brojects/action/GET_PROJECTS",
    },
  ],
  Array [
    Object {
      "projectId": "324041",
      "type": "brojects/action/GET_COLUMNS",
    },
  ],
  Array [
    Object {
      "columnId": "575404",
      "type": "brojects/action/GET_CARDS",
    },
  ],
]
`;

exports[`App mapDispatchToProps should handle onSubmit 1`] = `
Array [
  Array [
    Object {
      "columnId": "575404",
      "type": "brojects/action/GET_COLUMN_DATA",
    },
  ],
]
`;

exports[`App mapStateToProps should pass auth object 1`] = `
Object {
  "auth": Object {
    "some": "value",
  },
  "cards": undefined,
  "columnData": undefined,
  "columns": undefined,
  "repos": undefined,
}
`;
